File: 1.2 Patterns/19-hollow-diamond.cpp
#include <iostream>
using namespace std;
int main()
{
    int row = 5;
    for(int i = row ; i > 0 ; i--){
        for(int j = 0 ; j < i ; j++){
            cout<<"*";
        }
        for(int k = 0 ; k < 2*(row - i) ; k++){
            cout<<" ";
        }
        for(int j = 0 ; j < i ; j++){
            cout<<"*";
        }
        cout<<"\n";
    }
    for(int i = 1 ; i < row + 1 ; i++){
        for(int j = 0 ; j < i ; j++){
            cout<<"*";
        }
        for(int k = 0 ; k < 2*(row - i) ; k++){
            cout<<" ";
        }
        for(int j = 0 ; j < i ; j++){
            cout<<"*";
        }
        cout<<"\n";
    }
    return 0;
}

File: 1.2 Patterns/19-hollow-diamond.py
row = 5
for i in range (row, -1, -1):
    for j in range (0, i):
        print("*", end ="")
    for k in range (0, 2*(row-i) ):
        print(" ", end ="")
    for j in range (0, i):
        print("*", end ="")
    print("")
for i in range (1, row+1):
    for j in range (0, i):
        print("*", end ="")
    for k in range (2*(row - i)):
        print(" ", end ="")
    for j in range (0, i):
        print("*", end ="")
    print("")                             
                            
